lib/nimble_template/templates/template.ex

```diff
@@ -28,22 +28,24 @@ defmodule NimbleTemplate.Templates.Template do
   defp post_apply!(%Project{mix_project?: true} = project) do
     order_dependencies!()
     fetch_and_install_elixir_dependencies()
-    suppress_necessary_credo_warnings(project)
+    suppress_credo_warnings_for_base_project(project)
     format_codebase()
```

Comment:
@nvminhtue 

Which lines are you referring to, these two right?

```elixir
  order_dependencies!()
  fetch_and_install_elixir_dependencies()
```

Reply:
and ```elixir
format_codebase()
``` as well @liamstevens111

Reply:
These steps are already in `post_apply` and for the most part are distinct steps to me. The only thing I could think of for less repetition in this function would be to do as follows:

1. Extract 
```elixir
  order_dependencies()
  format_codebase()
```
To another function that all it does is call these two methods

2. Extract
```elixir
   fetch_and_install_elixir_dependencies()
   fetch_and_install_node_dependencies()
```
To another function that all it does is call these two methods.

So it would ultimately just be combining two function calls into one by using them in another function. No further pattern matching functions

Let me know if you have another idea @nvminhtue

Reply:
It's not essential overall, if those are calling in a common function, it could keep the could less repetitive, I think there's no matter in combining those 3 functions into a private function.

Reply:
yeah I did notice some repetition on it as well so good for spotting it. Maybe we can leave for now.